client/payment-details/payment-method/klarna/index.js

```diff
@@ -0,0 +1,146 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies.
+ */
+import PaymentDetailsPaymentMethodDetail from '../detail';
+
+/**
+ * Extracts and formats payment method details from a charge.
+ *
+ * @param {Object} charge The charge object.
+ * @return {Object}       A flat hash of all necessary values.
+ */
+const formatPaymentMethodDetails = ( charge ) => {
+	const { billing_details: billingDetails, payment_method: id } = charge;
+
+	const {
+		country: countryCode,
+		verified_name: verifiedName,
+		payment_method_category: paymentMethodCategory,
+		preferred_locale: preferredLocale,
+	} = charge.payment_method_details.klarna;
+
+	const { name, email, formatted_address: formattedAddress } = billingDetails;
+	// Use the full country name.
+	const country = wcSettings.countries[ countryCode ];
+
+	return {
+		verifiedName,
+		id,
+		name,
+		email,
+		country,
+		formattedAddress,
+		paymentMethodCategory,
+		preferredLocale,
+	};
+};
+
+/**
+ * Placeholders to display while loading.
+ */
+const paymentMethodPlaceholders = {
+	verifiedName: 'verified name placeholder',
+	fingerprint: 'fingerprint placeholder',
+	id: 'id placeholder',
+	name: 'name placeholder',
+	email: 'email placeholder',
+	formattedAddress: 'address placeholder',
+	country: 'country placeholder',
+};
+
+const KlarnaDetails = ( { charge = {}, isLoading } ) => {
+	const details = charge.payment_method_details
+		? formatPaymentMethodDetails( charge )
+		: paymentMethodPlaceholders;
+
+	const {
+		verifiedName,
+		id,
+		name,
+		email,
+		country,
+		formattedAddress,
+		paymentMethodCategory,
+		preferredLocale,
+	} = details;
+
+	// Shorthand for more readable code.
+	const Detail = PaymentDetailsPaymentMethodDetail;
+
+	return (
+		<div className="payment-method-details">
+			<div className="payment-method-details__column">
+				<Detail
+					isLoading={ isLoading }
+					label={ __( 'ID', 'woocommerce-payments' ) }
+				>
+					{ id }
+				</Detail>
+
+				<Detail
+					isLoading={ isLoading }
+					label={ __( 'Category', 'woocommerce-payments' ) }
+				>
+					{ paymentMethodCategory }
```

Comment:
I wonder if we should translate the paymentMethodCategory? According to Stripe [docs](https://stripe.com/docs/api/charges/object#charge_object-payment_method_details-klarna-payment_method_category) this can be `pay_later`, `pay_now`, `pay_with_financing`, `pay_in_installments `.

Reply:
See https://github.com/Automattic/woocommerce-payments/pull/7282#discussion_r1343950205